main.py

```diff
@@ -556,21 +556,46 @@ def populate_memes():
                     await delete_edit_timer(tmp, FEEDBACK_DEL_TIMER, call_msg=message)
                     return
                 if role_colour > 16777216 or role_colour < 0:
-                    tmp = await client.send_message(message.channel, "Usage: **!patreon #ff0000** is red\n**!patreon #00ff00** is green\netc...")
+                    tmp = await client.send_message(message.channel, "Usage: `!patreon #ff0000` is red\n`!patreon #00ff00` is green\netc...")
                     await delete_edit_timer(tmp, FEEDBACK_DEL_TIMER, call_msg=message)
                     return
                 role_name = "donor_c_" + str(message.author)
-                for r in message.author.roles: #check if user has a patreon role and delete it
+                for r in message.author.roles: #check if user has already a donor colour role and delete it
                     if r.name.startswith("donor_"):
                         await client.delete_role(message.server, r)
+                for r in message.server.roles: #check position of donor role
+                    if r.name.lower() == "donor":
+                        donor_pos = r.position
                 new_role = await client.create_role(
                     message.server, name=role_name, colour=discord.Colour(role_colour)
                 )
+                await client.move_role(message.server, new_role, donor_pos + 1)
```

Comment:
Shouldn't the "donor" position be determined *after* the new role has been added?